Softeq.XToolkit.PushNotifications/PushTokenSynchronizer.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using System.Threading.Tasks;
+using Softeq.XToolkit.Common.Extensions;
+using Softeq.XToolkit.Common.Interfaces;
+using Softeq.XToolkit.Common.Logger;
+using Softeq.XToolkit.PushNotifications.Abstract;
+
+namespace Softeq.XToolkit.PushNotifications
+{
+    public class PushTokenSynchronizer : IPushTokenSynchronizer
```

Comment:
Looks like this is `IPushTokenManager` (token delegate + token synchronizer)
@TimofeyBurak what do you think?

Reply:
You mean naming-wise? I'm not sure, "manager" to me sounds like a class that manages something, meaning controls something. This class does not control "PushToken", it only reacts to the changes made by other classes. If this class is controlling something - it's "PushToken Synchronisation", thus `IPushTokenSyncronizationManager` could be possible, but `IPushTokenSynchronizer` seems shorter with the same meaning)